lib/Sema/TypeCheckDecl.cpp

```diff
@@ -2855,6 +2880,13 @@ class DeclChecker : public DeclVisitor<DeclChecker> {
   void visitVarDecl(VarDecl *VD) {
     // Delay type-checking on VarDecls until we see the corresponding
     // PatternBindingDecl.
+
+    if (VD->isSettable(nullptr) &&
+        VD->getValueInterfaceType()->hasDynamicSelfType())
+      TC.diagnose(VD->getLoc(), diag::self_in_mutable_property);
+
+    diagnoseSelfTypedParameters(VD->getValueInterfaceType(),
```

Comment:
I don't think does anything for non-FuncDecls? You do want it for SubscriptDecls though

Reply:
You still get a (slightly obscure) error for non-func decls complaining about the setter.